app/src/main/java/org/mozilla/fenix/historymetadata/HistoryMetadataService.kt

```diff
@@ -81,11 +81,17 @@ class DefaultHistoryMetadataService(
     }
 
     override fun updateMetadata(key: HistoryMetadataKey, tab: TabSessionState) {
-        logger.debug("Updating metadata for tab $tab")
+        val lastAccess = tab.lastAccess
+        if (lastAccess == 0L) {
+            logger.debug("Not updating metadata for tab $tab - lastAccess=0")
+            return
+        } else {
+            logger.debug("Updating metadata for tab $tab")
+        }
 
         scope.launch {
             val viewTimeObservation = HistoryMetadataObservation.ViewTimeObservation(
-                viewTime = (System.currentTimeMillis() - tab.lastAccess).toInt()
+                viewTime = (System.currentTimeMillis() - lastAccess).toInt()
```

Comment:
Is this something that we should be preventing on the component itself? To share the validation between consumers?

Reply:
Maybe this is just a temporary fix to prevent the crash :)

Reply:
This should be the entry point where the `lastAccess` value stores the `viewTime` to the storage layer.

`lastAccess` can be 0 if the tab was created (with the context menu perhaps) and never accessed.